pkg/runtime/runtime.go

```diff
@@ -230,38 +239,31 @@ func (a *DaprRuntime) initRuntime(opts *runtimeOpts) error {
 
 	a.loadAppConfiguration()
 
-	// Register and initialize state stores
-	a.stateStoreRegistry.Register(opts.states...)
-	err = a.initState(a.stateStoreRegistry)
+	// Register and initialize name resolution for service discovery.
+	a.nameResolutionRegistry.Register(opts.nameResolutions...)
+	err = a.initNameResolution()
 	if err != nil {
-		log.Warnf("failed to init state: %s", err)
+		log.Warnf("failed to init name resolution: %s", err)
 	}
 
-	// Register and initialize pub/sub
+	a.exporterRegistry.Register(opts.exporters...)
 	a.pubSubRegistry.Register(opts.pubsubs...)
-	err = a.initPubSub()
-	if err != nil {
-		log.Warnf("failed to init pubsub: %s", err)
-	}
+	a.secretStoresRegistry.Register(opts.secretStores...)
+	a.stateStoreRegistry.Register(opts.states...)
+	a.bindingsRegistry.RegisterInputBindings(opts.inputBindings...)
+	a.bindingsRegistry.RegisterOutputBindings(opts.outputBindings...)
 
-	// Register and initialize exporters
-	a.exporterRegistry.Register(opts.exporters...)
-	err = a.initExporters()
+	go a.processComponents()
```

Comment:
If this finishes after lines 257, 161 or 262, this might be a problem, no? (since it's executed in a goroutine)

Reply:
@yaron2 sorry, i didn't get your question exactly.
The two goroutines don't affect each other, and communicate by the pending channel.